rust/gitxetcore/src/config/xet.rs

```diff
@@ -205,6 +220,112 @@ impl XetConfig {
         GitXetRepo::get_remote_urls(repo_path).unwrap_or_else(|_| vec!["".to_string()])
     }
 
+    // Check the cli override, the env var and Xetea
+    // in the listed order.
+    #[allow(dead_code)] // only dead in unit tests, add to avoid clippy complains
+    async fn config_cas(&self) -> Result<String, ConfigError> {
+        let mut cas_endpoint = String::new();
+        if let Some(over) = &self.overrides {
+            if let Some(cas) = &over.cas {
+                cas_endpoint = cas.clone();
+            }
+        }
+
+        if cas_endpoint.is_empty() {
+            if let Some(envvar) = std::env::var_os(XET_CAS_SERVER_ENV_VAR)
+                .as_ref()
+                .and_then(|osstr| osstr.to_str())
+            {
+                cas_endpoint = envvar.to_owned();
+            }
+        }
+
+        if cas_endpoint.is_empty() {
+            let urls = self.remote_urls.clone();
+            // No CAS endpoint configured from overrides, we try to retrieve it from Xetea.
+            let maybe_cas = tokio_par_for_each(urls, 10, |remote, _| {
```

Comment:
This seems a little heavy, as every remote is queried every time.  If one random host times out or isn't valid, then it would hang the whole process... 

I think it would be better to do something like this: 

```
let urls = self.remote_urls.clone();

let mut query_attempts = urls.map(|(remote, _)|   tokio.spawn(async move |remote| {  get_cas_endpoint_from_git_remote(...).await }).collect::<VecDeque<_>>();

while let Some(jh) = query_attempts.pop_front() {
  if let Ok(res) = jh.await? {
     cas = ...
     break;
    } 
   }
 
   // Clean up
  while let Some(jh) = query_attempts.pop_front() {
     jh.abort()
     }
  }
  ```

Reply:
I guess this would then use the first remote?

Reply:
Yeah.... Not sure that this is better?
Though, get_cas_endpoint_from_git_remote should retry failures if it does not already.

Reply:
Changed to use the first valid response from Xetea.